questions/services/forecasts.py

```
@@ -34,28 +35,75 @@
     now = timezone.now()
     post = question.get_post()
+    source = source or ""
+
+    # delete all future-dated predictions, as this one will override them
+    Forecast.objects.filter(question=question, author=user, start_time__gt=now).delete()
+
+    # if the forecast to be created is for a multiple choice question during a grace
+    # period, we need to agument the forecast accordingly (possibly preregister)
+    if question.type == Question.QuestionType.MULTIPLE_CHOICE:
+        if not probability_yes_per_category:
+            raise ValueError("probability_yes_per_category required for MC questions")
```

Ruff (0.14.11)

Avoid specifying long messages outside the exception class

(TRY003)